### DISCUSSION TITLE
Can somebody explain further difference of using Essential and Elevated?

### DISCUSSION THREAD

**Discussion by jisungyoon019** [plhery/node-twitter-api-v2#235] [Q&A] [answered]
Hi, I would like for somebody to guide me on how I am able to use Twitter Dev API to perform tweet actions in nodejs Visual Studio.

I was using tweepy in python for a while but I think it's slow when finding new tweets, it takes like 5 to 6 seconds delay even in Stream client.

Here are my questions, I did read all the docs but I'm still confused...maybe I'm slow..?
![Help](https://user-images.githubusercontent.com/102338546/160043835-861bc5f9-3a11-4675-946f-450825693c9c.png)

I hope all my questions in the image gets covered. Thanks everyone!

**Comment by jisungyoon019** [comment]
Hi I was able to make it work by using this:
const client = new TwitterApi({
    appKey: CONSUMER_KEY,
    appSecret: CONSUMER_SECRET,
    accessToken: ACCESS_TOKEN, 
    accessSecret: ACCESS_TOKEN_SECRET, 
  });
  
But I still need more enlightenment for my questions from the image above...yes sorry I'm a complete beginner with a little background on javascript and python

**Comment by plhery** [comment]
- Essentials gives you only access to v2 endpoints, indeed https://developer.twitter.com/en/docs/twitter-api/getting-started/about-twitter-api
- you can in the twitter apps settings change whether your app is read-only, read-write, read-write-and-DM ; by default it's read-only (you can fetch tweets but not post tweets) https://developer.twitter.com/en/docs/apps/app-permissions
- If I'm not mistaken, you can indeed use any auth method you'd like

- you can always try things, if you don't have the right to do something, you'll simply have a clear error

**Comment by alkihis** [comment, marked as answer]
As @PLhery said, Essentials level only gives you access to v2 endpoints. But you're not restricted on authentication method (OAuth 1/2, App only, etc.)

If you want to send tweets for your accounts, or get your own tweets, here's a quick guide.

- Create a client linked to your user
```ts
const client = new TwitterApi({
  // appKey/appSecret are identifiers for your application on Twitter
  appKey: '', // < Here's the "consumer key" available on the dev portal
  appSecret: '', // < Here's the "consumer secret" available on the dev portal

  // access keys are special keys defining an association between your application and a Twitter account
  accessToken: '', // < Here's your "access token", which can be generated under consumer keys
  accessSecret: '', // < Here's the secret associated under access token
})
```

- List your own tweets
```ts
const me = await client.v2.me()

// Get your user ID from your user object
const myUserId = me.data.id

// Get my tweets using your user ID
const myTweets = await client.v2.userTimeline(meUserId)

// Iterate over fetched tweets
for (const tweet of myTweets) {
  console.log('Tweet', tweet.id, ', text:', tweet.text)
}

// If there's more tweets to go, you can fetch the next page
if (!myTweets.done) {
  const nextPage = await myTweets.next()

  for (const tweet of nextPage) {
    console.log('Page 2 ; Tweet', tweet.id, ', text:', tweet.text)
  }

  // etc...
}
```

- Send a tweet with your account
```ts
await client.v2.tweet('hello!')
```

**Reply by jisungyoon019** [reply]
Thank you for the answers! I and I learned a lot as a beginner. I will also post a new thread based on some observation on Stream Listener performance anytime soon! Thanks again!
